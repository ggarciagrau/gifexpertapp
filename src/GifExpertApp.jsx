import { useState } from "react";
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dogo']);

    const onAddCategory = (value) => {

        if (categories.includes(value)) return;

        setCategories([value, ...categories])
        // setCategories(['Birbo', ...categories]);
    };

    return (
        <>
            {/* Title */}
            <h1>GifExpertApp</h1>

            {/* Input */}
            <AddCategory
                onNewCategory={onAddCategory}
            // setCategories={setCategories}
            />

            {/* Gif list */}
            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category}
                    />
                ))
            }

            {/* Gif item */}
        </>
    );
}