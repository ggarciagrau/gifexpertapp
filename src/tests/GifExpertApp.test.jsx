import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../GifExpertApp";

describe('GifExpertApp tests', () => {

    test("Initial render", () => {

        const { container } = render(<GifExpertApp />);

        expect(screen.getByText('GifExpertApp')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    test("Must add category", () => {

        const { container } = render(<GifExpertApp />);

        const input = screen.getByRole('textbox');

        fireEvent.change(input, { target: { value: 'Birbo' } });
        fireEvent.submit(screen.getByRole('form'));

        expect(screen.getByText('Birbo')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

    test("Mustn't add repeated category", () => {

        const { container } = render(<GifExpertApp />);

        const input = screen.getByRole('textbox');

        fireEvent.change(input, { target: { value: 'Dogo' } });
        fireEvent.submit(screen.getByRole('form'));

        fireEvent.change(input, { target: { value: 'Dogo' } });
        fireEvent.submit(screen.getByRole('form'));

        expect(screen.getAllByText('Dogo').length).toBe(1);
        expect(screen.getByText('Dogo')).toBeTruthy();
        expect(container).toMatchSnapshot();
    });
});