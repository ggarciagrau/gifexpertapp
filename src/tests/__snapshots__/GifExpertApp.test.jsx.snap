// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests Initial render 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <div>
    <form
      aria-label="form"
    >
      <input
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </div>
  <h3>
    Dogo
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`GifExpertApp tests Must add category 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <div>
    <form
      aria-label="form"
    >
      <input
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </div>
  <h3>
    Birbo
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Dogo
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`GifExpertApp tests Mustn't add repeated category 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <div>
    <form
      aria-label="form"
    >
      <input
        placeholder="Search gifs..."
        type="text"
        value=""
      />
    </form>
  </div>
  <h3>
    Dogo
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
